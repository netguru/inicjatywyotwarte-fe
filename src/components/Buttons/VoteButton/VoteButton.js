import React from 'react'
import ArrowDropUpIcon from '@material-ui/icons/ArrowDropUp'
import styled from 'styled-components'

const StyledButtonVotesCounter = styled.span`
  font-weight: 300;
  height: 14px; /* hack - to make this font look better with */
  color: inherit;
  padding-left: 0.5rem;
`

const UpvoteButton = styled.button`
  color: ${props => props.theme.fontColorDark};
  font-size: 0.85rem;
  font-weight: 500;
  height: 40px;
  width: 100%;
  border-radius: ${props => props.theme.borderRadiusBig};
  cursor: pointer;
  outline: none;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0;
  border: 1px solid;
  border-color: ${props =>
    props.alreadyUpvoted
      ? props.theme.addInitiativeButtonBorderColor
      : props.theme.resourceTileLayoutColor
  };
  transition: ${props => props.theme.buttonTransition};

  &:hover {
    color: ${props => props.theme.lightBackgroundColor};
    background-color: ${props => props.theme.addInitiativeButtonBorderColor};
    border-color: ${props => props.theme.addInitiativeButtonBorderColor};

    > * {
      color: ${props => props.theme.lightBackgroundColor};
    }
  }
`
UpvoteButton.displayName = 'UpvoteButton';

export default function VoteButton ({
  votesCount,
  handleVote,
  alreadyUpvoted
}) {
  const buttonContent = alreadyUpvoted ? 'Pomogło' : 'Pomocne'

  return (
    <>
      <UpvoteButton
        onClick={handleVote}
        aria-label='głosuj'
        alreadyUpvoted={alreadyUpvoted}
      >
        <ArrowDropUpIcon />
        {buttonContent}&nbsp;
        <StyledButtonVotesCounter>({votesCount})</StyledButtonVotesCounter>
      </UpvoteButton>
    </>
  )
}
