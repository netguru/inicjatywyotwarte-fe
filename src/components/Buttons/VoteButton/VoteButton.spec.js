import React from 'react'
import VoteButton from './VoteButton'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const baseProps = {
  votesCount: 40,
  handleVote: jest.fn(),
  alreadyUpvoted: false,
};

describe('VoteButton', () => {
  test('renders upvote button', () => {
    const wrapper = shallow(<VoteButton {...baseProps} />);
    expect(wrapper.find('UpvoteButton').length).toEqual(1);
    expect(wrapper.find('UpvoteButton').text()).toEqual('Pomocne\u00a0(40)');
  });

  test('renders upvoted button', () => {
    const upvotedProps = {
      ...baseProps,
      alreadyUpvoted: true
    };

    const wrapper = shallow(<VoteButton {...upvotedProps} />);
    expect(wrapper.find('UpvoteButton').length).toEqual(1);
    expect(wrapper.find('UpvoteButton').text()).toEqual('PomogÅ‚o\u00a0(40)');
  });

  test('match snapshot', () => {
    const wrapper = shallow(<VoteButton {...baseProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


