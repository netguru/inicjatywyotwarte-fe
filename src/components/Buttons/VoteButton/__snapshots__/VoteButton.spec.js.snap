// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteButton match snapshot 1`] = `
<Fragment>
  <UpvoteButton
    alreadyUpvoted={false}
    aria-label="głosuj"
    onClick={[MockFunction]}
  >
    <ArrowDropUpIcon />
    Pomocne
     
    <styled.span>
      (
      40
      )
    </styled.span>
  </UpvoteButton>
</Fragment>
`;
